.github/workflows/ci.yaml

```diff
@@ -178,6 +178,73 @@ jobs:
           path: coverage.xml
           retention-days: 7
 
+  test-slurm:
+    runs-on: ubuntu-latest
+    services:
+      slurm:
+        image: autosubmit/slurm-openssh-container:25-05-0-1
+        ports:
+          - 2222:2222
+        options: --rm -it --cgroupns=host --privileged --volume /sys/fs/cgroup:/sys/fs/cgroup:rw --hostname slurmctld --name slurm-container
+
+    strategy:
+      matrix:
+        python-version: ["3.9"]
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Copy SSH access key
+        run: |
+          docker cp slurm-container:/root/.ssh/container_root_pubkey /tmp/container_root_pubkey
+          chmod 600 /tmp/container_root_pubkey
+
+      - name: Configure SSH host
+        run: |
+          sudo bash -c 'mkdir /home/runner/.ssh/'
+          sudo bash -c 'echo -e $"Host localDocker\n  HostName localhost\n  User root\nStrictHostKeyChecking no\n UserKnownHostsFile /dev/null\n  IdentityFile /tmp/container_root_pubkey\n  Port 2222\n   ForwardX11 yes" >> /home/runner/.ssh/config'
+
+      - name: Run sinfo over SSH on SLURM container
+        run: |
+          ssh localDocker -p 2222 -i /tmp/container_root_pubkey sinfo
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install system dependencies
+        run: sudo apt-get install -y curl git graphviz rsync
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip packaging
+          pip install --upgrade -e .[all]
+
+      - name: Install test dependencies
+        run: |
+          pip install pytest pytest-cov pytest-xdist
```

Comment:
Ah, I think these are not needed. `[all]` should be including these.

![image](https://github.com/user-attachments/assets/d78ffc00-7dc5-440b-838c-33597dd31062)